code/game/jobs/job/civilian_chaplain.dm

```diff
@@ -12,151 +12,152 @@
-				if("homosexuality")
-					B.name = "Guys Gone Wild"
-				if("moroz holy tribunal","moroz","holy tribunal")
-					B.name = "Tribunal Codex"
-				//if("lol", "wtf", "gay", "penis", "ass", "poo", "badmin", "shitmin", "deadmin", "cock", "cocks")
-				//	B.name = pick("Woodys Got Wood: The Aftermath", "War of the Cocks", "Sweet Bro and Hella Jef: Expanded Edition")
-				//	H.setBrainLoss(100) // starts off retarded as fuck
-				if("science")
-					B.name = pick("Principle of Relativity", "Quantum Enigma: Physics Encounters Consciousness", "Programming the Universe", "Quantum Physics and Theology", "String Theory for Dummies", "How To: Build Your Own Warp Drive", "The Mysteries of Bluespace", "Playing God: Collector's Edition")
-				else
-					B.name = "The Holy Book of [new_religion]"
-			feedback_set_details("religion_name","[new_religion]")
+	spawn(0)
+		var/religion_name = "Christianity"
+		var/new_religion = sanitize(input(H, "You are the crew services officer. Would you like to change your religion? Default is Christianity, in SPACE.", "Name change", religion_name), MAX_NAME_LEN)
 
-		spawn(1)
-			var/deity_name = "Space Jesus"
-			var/new_deity = sanitize(input(H, "Would you like to change your deity? Default is Space Jesus.", "Name change", deity_name), MAX_NAME_LEN)
+		if (!new_religion)
+			new_religion = religion_name
 
-			if ((length(new_deity) == 0) || (new_deity == "Space Jesus") )
-				new_deity = deity_name
-			B.deity_name = new_deity
+		switch(lowertext(new_religion))
+			if("christianity")
+				B.name = pick("The Holy Bible","The Dead Sea Scrolls")
+			if("satanism")
+				B.name = "The Unholy Bible"
+			if("cthulu")
+				B.name = "The Necronomicon"
+			if("islam")
+				B.name = "Quran"
+			if("scientology")
+				B.name = pick("The Biography of L. Ron Hubbard","Dianetics")
+			if("chaos")
+				B.name = "The Book of Lorgar"
+			if("imperium")
+				B.name = "Uplifting Primer"
+			if("toolboxia")
+				B.name = "Toolbox Manifesto"
+			if("homosexuality")
+				B.name = "Guys Gone Wild"
+			if("moroz holy tribunal","moroz","holy tribunal")
+				B.name = "Tribunal Codex"
+			//if("lol", "wtf", "gay", "penis", "ass", "poo", "badmin", "shitmin", "deadmin", "cock", "cocks")
+			//	B.name = pick("Woodys Got Wood: The Aftermath", "War of the Cocks", "Sweet Bro and Hella Jef: Expanded Edition")
+			//	H.setBrainLoss(100) // starts off retarded as fuck
+			if("science")
+				B.name = pick("Principle of Relativity", "Quantum Enigma: Physics Encounters Consciousness", "Programming the Universe", "Quantum Physics and Theology", "String Theory for Dummies", "How To: Build Your Own Warp Drive", "The Mysteries of Bluespace", "Playing God: Collector's Edition")
+			else
+				B.name = "The Holy Book of [new_religion]"
+		feedback_set_details("religion_name","[new_religion]")
 
-			var/accepted = 0
-			var/outoftime = 0
-			spawn(200) // 20 seconds to choose
-				outoftime = 1
-			var/new_book_style = "Bible"
+	spawn(1)

```

Comment:
Considering the size of the block under spawn(), might be worth refactoring into a separate proc and putting it into a CALLBACK instead. Minor optimizations.